src/Dto/AddProductBundleToCartDto.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+ */
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusProductBundlePlugin\Dto;
+
+use BitBag\SyliusProductBundlePlugin\Command\AddProductBundleItemToCartCommand;
+use BitBag\SyliusProductBundlePlugin\Command\ProductCodeAwareInterface;
+use BitBag\SyliusProductBundlePlugin\Entity\ProductInterface;
+use Doctrine\Common\Collections\ArrayCollection;
+use Sylius\Component\Order\Model\OrderInterface;
+use Sylius\Component\Order\Model\OrderItemInterface;
+
+final class AddProductBundleToCartDto implements AddProductBundleToCartDtoInterface, ProductCodeAwareInterface
+{
+    /** @var OrderInterface */
+    private $cart;
+
+    /** @var OrderItemInterface */
+    private $cartItem;
+
+    /** @var ProductInterface */
+    private $product;
+
+    /** @var ArrayCollection */
+    private $productBundleItems;
+
+    public function __construct(
+        OrderInterface $cart,
+        OrderItemInterface $cartItem,
+        ProductInterface $product
+    ) {
+        $this->cart = $cart;
+        $this->cartItem = $cartItem;
+        $this->product = $product;
+        $this->productBundleItems = new ArrayCollection();
+
+        $this->processProductBundleItems();
```

Comment:
Are you sure that DTO should do any logic like that?

Reply:
I've created DTO factory method inside `OrderItemController` and moved processing bundleItems there.